test/e2e/e2e_suite_test.go

```diff
@@ -300,3 +301,26 @@ func updateCalico(config *clusterctl.E2EConfig, calicoYaml, calicoInterface stri
 	err = os.WriteFile(calicoYaml, yamlOut, 0600)
 	Expect(err).ToNot(HaveOccurred(), "Cannot print out the update to the file")
 }
+
+// postNamespaceCreated is a hook function that can be called after creating
+// a namespace, it creates the needed bmhs in the namespace hosting the cluster.
+func postNamespaceCreated(clusterProxy framework.ClusterProxy, clusterNamespace string) {
```

Comment:
nit: Perhaps better to name it after what it is doing?
```suggestion
func createBMHsInNamespace(clusterProxy framework.ClusterProxy, clusterNamespace string) {
```